applications/GeoMechanicsApplication/custom_constitutive/incremental_linear_elastic_interface_law.cpp

```diff
@@ -40,6 +40,23 @@ Vector& GeoIncrementalLinearElasticInterfaceLaw::GetValue(const Variable<Vector>
         rValue = mPreviousRelativeDisplacement;
     } else if (rThisVariable == CAUCHY_STRESS_VECTOR) {
         rValue = mPreviousTraction;
+    } else {
+        KRATOS_ERROR << "Can't get value of " << rThisVariable.Name() << ": unsupported variable\n";
+    }
+
+    return rValue;
+}
+
+Matrix& GeoIncrementalLinearElasticInterfaceLaw::CalculateValue(ConstitutiveLaw::Parameters& rParameterValues,
+                                                                const Variable<Matrix>& rThisVariable,
+                                                                Matrix& rValue)
+{
+    if (rThisVariable == CONSTITUTIVE_MATRIX) {
+        const auto& r_properties = rParameterValues.GetMaterialProperties();
+        rValue                   = MakeConstitutiveMatrix(r_properties[INTERFACE_NORMAL_STIFFNESS],
+                                                          r_properties[INTERFACE_SHEAR_STIFFNESS]);
+    } else {
+        KRATOS_ERROR << "Can't calculate value of " << rThisVariable.Name() << ": unsupported variable\n";
```

Comment:
There is a subtle difference between the error messages: the first one refers to _getting_ the value, whereas the second one refers to _calculating_ it. I agree with you that introducing such generic error reporting functions do make sense when we start to implement overloads of `GetValue` and `CalculateValue`.

As discussed, I won't take any further action yet.

Reply:
Just a comment, raising errors when printing can be problematic. Let's imagine that you have a mesh with two kind of FE. In one FE you are able to print one variable but in the other raises an error. The code will crash even though for some FE the variable is correct... This is why we do not usually raise error if the variable is not found, we just skip overriding the `rValue`

Reply:
Dear @AlejandroCornejo, thank you very much for sharing your thoughts on throwing exceptions from `GetValue` and `CalculateValue`. For your information, we intend to use these member functions not exclusively for printing, but also as part of the actual analysis. In that case, I reckon we would actually like to receive an error when a variable is not present rather than silently proceeding. Your concern about some variable being available for one kind of element, but not for another one can be addressed by grouping elements in different model parts, and only output data for model parts that are expected to have a certain variable.
My major concern of silently proceeding when an element doesn't have some variable, is that problems may go undetected for a long time. I'd rather crash early. If things turn out to be problematic, we will consider to raise a warning instead.